cs/tech.go

```diff
@@ -404,26 +668,24 @@ func (t *Tech) GetPlayerCost(techLevels TechLevel, spec MiniaturizationSpec, cos
 		miniaturizationFactor = 1 - miniaturization
 	}
 
-	// apply any tech cost offsets
+	// planetary items don't get miniaturization
+	if t.Category == TechCategoryPlanetary || t.Category == TechCategoryTerraforming || t.Category == TechCategoryPlanetaryDefense || t.Category == TechCategoryPlanetaryScanner {
+		miniaturizationFactor = 1
+	} 
+
+	// apply any tech cost offsets to our item cost
 	cost := t.Cost
-	switch t.Category {
-	case TechCategoryEngine:
-		cost = cost.MultiplyFloat64(1 + costOffset.Engine)
-	case TechCategoryBeamWeapon:
-		cost = cost.MultiplyFloat64(1 + costOffset.BeamWeapon)
-	case TechCategoryBomb:
-		cost = cost.MultiplyFloat64(1 + costOffset.Bomb)
-	case TechCategoryTorpedo:
-		cost = cost.MultiplyFloat64(1 + costOffset.Torpedo)
-	case TechCategoryTerraforming:
-		cost = cost.MultiplyFloat64(1 + costOffset.Terraforming)
+	var highestCostMulti float64
+	for tag := range t.Tags {
+		highestCostMulti = math.Min(1+costOffset[tag], highestCostMulti)
```

Comment:
IDK whether to implement category cost reduction stacking as multiplicative, additive or just "take biggest boost". 
OG Stars! appears to have done all its bonuses either by tech category or as a global boost, so we don't really have any baseline to work off of.
(Personally, I'm leaning towards "biggest boost" as it makes the most sense and ensures we won't have clashes with giving all X type items a special tag)